tests/test_elevation_motion.py

```diff
@@ -0,0 +1,392 @@
+            The target position.
+        velocity: `float`
+            The velocity for the motion.
+        expected_duration: `float`
+            The expected duration.
+        motion_state: `MotionState`
+            The commanded MotionState.
+        """
+        duration = self.elevation_motion.set_target_position_and_velocity(
+            start_tai=start_tai,
+            end_position=target_position,
+            crawl_velocity=velocity,
+            motion_state=motion_state,
+        )
+        self.assertEqual(expected_duration, duration)
+
+    async def verify_elevation_motion_position(
+        self, tai, expected_position, expected_motion_state
+    ):
+        """Verify the position of the ElevationMotion at the given TAI
+        time.
+
+        Parameters
+        ----------
+        tai: `float`
+            The TAI time to compute the position for.
+        expected_position: `float`
+            The expected position at the given TAI time.
+        expected_motion_state: `float`
+            The expected motion state at the given TAI time.
+        """
+        position, motion_state = self.elevation_motion.get_position_and_motion_state(
+            tai
+        )
+        self.assertAlmostEqual(expected_position, position)
+        self.assertEqual(expected_motion_state, motion_state)
+
+    async def test_move_zero_ten(self):
+        """Test the ElevationMotion when moving from position 0 to
+        position 10.
+        """
+        start_position = 0.0
+        start_tai = _start_tai
+        min_position = _MIN_POSITION
+        max_position = _MAX_POSITION
+        max_speed = math.radians(3.5)
+        target_position = math.radians(10.0)
+        velocity = math.radians(3.5)
+        expected_duration = (target_position - start_position) / velocity
+        await self.prepare_elevation_motion(
+            start_position=start_position,
+            min_position=min_position,
+            max_position=max_position,
+            max_speed=max_speed,
+            start_tai=start_tai,
+        )
+        await self.verify_elevation_motion_duration(
+            start_tai=start_tai,
+            target_position=target_position,
+            velocity=velocity,
+            expected_duration=expected_duration,
+            motion_state=MotionState.MOVING,
+        )
+        await self.verify_elevation_motion_position(
+            tai=_start_tai + 1.0,
+            expected_position=math.radians(3.5),
+            expected_motion_state=MotionState.MOVING,
+        )
+        await self.verify_elevation_motion_position(
+            tai=_start_tai + 2.5,
+            expected_position=math.radians(8.75),
+            expected_motion_state=MotionState.MOVING,
+        )
+        await self.verify_elevation_motion_position(
+            tai=_start_tai + 3.0,
+            expected_position=math.radians(10.0),
+            expected_motion_state=MotionState.STOPPED,
+        )
+
+    async def test_move_ten_zero(self):
```

Comment:
Please consider `check_move(self, start_pos, end_pos)` and then calling it with a few different pairs of values, preferably including the upper and lower limits.

You'd want to assert that the values are both in range, but the rest of the code would probably be similar, perhaps with the intermediate checks based on how long the move will take.

Reply:
My unit tests actually already contain methods exactly for that. I admit that the tests appear to be very verbose and I'll think of a way to make them more compact.